File: 5-Permutations/permutations.py
test_1 = 5;
test_2 = 3;
test_3 = 4;
test_4 = 1;
test_5 = 2;



def permutations(n):
	if n == 1:
		return [1]
	if n < 4:
		return "NO SOLUTION"
	
	odds = []
	evens = []

	for i in range(n):
		if i % 2 == 0:
			evens.append(i + 1)
		else:
			odds.append(i + 1)
	return odds + evens


print(permutations(test_1)) # [2, 4, 1, 3, 5] 
print(permutations(test_2)) # NO SOLUTION
print(permutations(test_3)) # [2, 4, 1, 3]
print(permutations(test_4)) # [1]
print(permutations(test_5)) # NO SOLUTION

File: 5-Permutations/permutations.js
const test1 = 5;
const test2 = 3;
const test3 = 4;
const test4 = 1;
const test5 = 2;

function permutations(num) {
  if (num === 1) {
    return [1];
  }

  if (num < 4) {
    return "NO SOLUTION";
  }
  let even = [];
  let odds = [];
  for (let i = 1; i <= num; i++) {
    if (i % 2 === 0) {
      even.push(i);
    } else {
      odds.push(i);
    }
  }

  return even.concat(odds);
}

console.log(permutations(test1)); // [2, 4, 1, 3, 5]
console.log(permutations(test2)); // NO SOLUTION
console.log(permutations(test3)); // [2, 4, 1, 3]
console.log(permutations(test4)); // [1]
console.log(permutations(test5)); // NO SOLUTION
